include/stlcache/stlcache.hpp

```diff
@@ -460,9 +463,11 @@ namespace stlcache {
          *  
          */
         void clear() throw() {
-            _storage.clear();
-            _policy->clear();
-            this->_currEntries=0;
+            mtx.lock();
+                _storage.clear();
+                _policy->clear();
+                this->_currEntries=0;
+            mtx.unlock();
```

Comment:
And with RAII mutex it will be automatically released, when goes out of scope. In your cahse it may not be released do to exception, thrown earlier.